mdtraj/core/topology.py

```diff
@@ -88,12 +88,17 @@ def _topology_from_subset(topology, atom_indices):
 
     for chain in topology._chains:
         newChain = newTopology.add_chain()
+        previous_residue = None
         for residue in chain._residues:
             resSeq = getattr(residue, 'resSeq', None) or residue.index
-            newResidue = newTopology.add_residue(residue.name, newChain,
-                                                 resSeq)
             for atom in residue._atoms:
                 if atom.index in atom_indices:
+                    if not residue == previous_residue:
+                        newResidue = newTopology.add_residue(residue.name, newChain,
+                                                             resSeq)
+                        newResidue.segment_id = residue.segment_id
```

Comment:
instead of setting the attribute on the new object, `segment_id` should be added as a new argument to the `Topology.add_residue` factory function, to parallel the `resSeq` record, I think.

Also, what is the reason for moving this logic inside the for loop in this PR?